src/map/skill.cpp

```diff
@@ -20896,32 +20893,18 @@ int32 skill_autospell(map_session_data *sd, uint16 skill_id)
 		maxlv = 10; //Soul Linker bonus. [Skotlex]
 	else
 		maxlv = skill_lv / 2; // Half of Autospell's level unless player learned a lower level (capped below)
```

Comment:
We could confirm now that in renewal, the level should also be determined per attack and not during casting of Hindsight, so this logic needs to be moved to the other part as well. Then the code here can just be shared between pre-re and renewal.